tests/contract_testcases/ewasm/call/call_02.dot

```diff
@@ -13,6 +13,7 @@ strict digraph "tests/contract_testcases/ewasm/call/call_02.sol" {
 	diagnostic [label="found contract ‘x’\nlevel Debug\ntests/contract_testcases/ewasm/call/call_02.sol:2:9-20"]
 	diagnostic_14 [label="destructure variable ‘s‘ has never been used\nlevel Warning\ntests/contract_testcases/ewasm/call/call_02.sol:4:23-24"]
 	diagnostic_15 [label="destructure variable ‘bs‘ has never been used\nlevel Warning\ntests/contract_testcases/ewasm/call/call_02.sol:4:39-41"]
+	diagnostic_16 [label="function can be declared ‘view’\nlevel Warning\ntests/contract_testcases/ewasm/call/call_02.sol:3:13-49"]
```

Comment:
Is this right? The function makes an external call.

Reply:
This is what the PR is trying to fix. solc says this is permitted.

Reply:
The question is whether the called contract modifies state. We don't know that, because `.call()` is without the abi.

solc assumes it views state, but does not modify it. This is useful because then call be used in rpc/readonly only mode. That means no transaction is required: this means it is free and fast.

Making users pay for transactions because you might modify state might not be appreciated.

On Substrate call is explicitly disallowed for rpc calls.